File: Simulation/CPP/design_parking_system.cpp
class ParkingSystem {
    public:
        int ps[4];
        ParkingSystem(int big, int medium, int small) {
            ps[1] = big, ps[2] = medium, ps[3] = small;
        }

        bool addCar(int carType) {
            return ps[carType]-- > 0 ? true : false;
        }
    };

    /**
     * Your ParkingSystem object will be instantiated and called as such:
     * ParkingSystem* obj = new ParkingSystem(big, medium, small);
     * bool param_1 = obj->addCar(carType);
     */

File: Simulation/Python/design_parking_system.py
class ParkingSystem:

    def __init__(self, big: int, medium: int, small: int):
        self.ps = [0, big, medium, small]


    def addCar(self, carType: int) -> bool:
        if self.ps[carType]:
            self.ps[carType] -= 1
            return True
        return False


# Your ParkingSystem object will be instantiated and called as such:
# obj = ParkingSystem(big, medium, small)
# param_1 = obj.addCar(carType)